flatpak-proxy.c

```diff
@@ -2940,7 +3004,54 @@ side_in_cb (GSocket *socket, GIOCondition condition, gpointer user_data)
               got_buffer_from_side (side, buffer);
 
               if (found_auth_end)
-                client->authenticated = TRUE;
+                {
+                  /* Immediately send the outgoing buffer */
+                  GSocketConnection *connection = client->bus_side.connection;
+                  GSocket *bus_socket = g_socket_connection_get_socket (connection);
+                  send_outgoing_buffers (bus_socket, &client->bus_side);
+
+                  client->auth_state = AUTH_WAITING_FOR_BACKLOG;
+                  check_pending_auth_lines (client);
+                }
+            }
+          else
+            {
+              buffer_unref (buffer);
+            }
+        }
+      else if (client->auth_state == AUTH_WAITING_FOR_BACKLOG)
+        {
+          if (buffer->pos > 0)
+            {
+              if (side == &client->bus_side)
+                {
+                  guint8 *line_start = buffer->data;
+                  guint8 *line_end = NULL;
+                  gsize extra_data = 0;
+
+                  while ((line_end = find_auth_line_end (line_start, buffer->pos)))
+                    {
+                      line_start = line_end + strlen (AUTH_LINE_SENTINEL);
+
+                      client->auth_server_replies++;
+                      if (client->auth_server_replies == client->auth_server_backlog)
+                        {
+                          buffer->size = line_start - buffer->data;
+                          extra_data = buffer->pos - buffer->size;
+
+                          /* We may have gotten some extra data which is not part of
+                            the auth handshake, keep it for the next iteration. */
+                          if (extra_data > 0)
+                            side->extra_input_data = g_bytes_new (line_start, extra_data);
+
+                          break;
+                        }
+                    }
+                }
+
+              got_buffer_from_side (side, buffer);
+
+              check_pending_auth_lines (client);
```

Comment:
When we set AUTH_COMPLETE when we get a bus message there may be outstanding extra_input_data from the auth message on the client input  side, but unless we get some action on the input fd we will not read it. So we need to manually trigger a read in this case.